test/framework/clusterctl/e2e_config.go

```diff
@@ -245,6 +250,102 @@ type Files struct {
 	TargetName string `json:"targetName,omitempty"`
 }
 
+// ResolveReleases converts release markers to release version.
+func (c *E2EConfig) ResolveReleases() error {
+	for i := range c.Providers {
+		provider := &c.Providers[i]
+		for j := range provider.Versions {
+			version := &provider.Versions[j]
+			if version.Type == "url" {
+				releaseMarker := strings.TrimLeft(strings.TrimRight(version.Name, "}"), "{")
+				if !strings.EqualFold(version.Name, releaseMarker) {
+					ver, err := resolveReleaseMarker(releaseMarker)
+					if err != nil {
+						return fmt.Errorf("failed resolving release path %s with error: %v", version.Name, err)
+					}
+					ver = "v" + ver
+					version.Value = strings.Replace(version.Value, version.Name, ver, 1)
+					version.Name = ver
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// resolveReleaseMarker resolves releaseMarker string to verion string e.g.
+// - Resolves "{go://sigs.k8s.io/cluster-api@v1.0}" to stable minor release of v.1.0.
+// - Resolves "{go://sigs.k8s.io/cluster-api@latest-v1.0}" to latest minor release of v.1.0 including rc and pre releases.
+func resolveReleaseMarker(releaseMarker string) (string, error) {
+	parts := strings.Split(releaseMarker, ":")
+	if len(parts) < 1 {
+		return "", errors.Errorf("Invalid release url")
+	}
+
+	host := parts[0]
+	gomoduleParts := strings.Split(parts[1], "@")
+	if len(gomoduleParts) < 2 {
+		return "", errors.Errorf("Invalid release url, go module or version missing")
+	}
+	gomodule := gomoduleParts[0]
+	version := strings.ReplaceAll(gomoduleParts[1], "v", "")
+	includePrereleases := false
+	parsedVersion := strings.Split(version, "-")
+	if len(parsedVersion) > 0 {
+		if len(parsedVersion) > 1 {
+			version = parsedVersion[1]
+			if parsedVersion[0] == "latest" {
+				includePrereleases = true
+			}
+		}
+	} else {
+		return "", errors.Errorf("Invalid release url")
+	}
```

Comment:
Maybe make use of `strings.HasPrefix` and `strings.TrimPrefix` instead to make it more readable what's happening.

The error handling could be done via the `semver.Parse` then.

```suggestion
	if strings.HasPrefix(parsedVersion, "latest-") {
		includePrereleases = true
	}
	version := strings.TrimPrefix(parsedVersion, "latest-")
```

Reply:
May be also worth using above or below.